_includes/aiml-sidebar.html

```diff
@@ -0,0 +1,129 @@
+        <br>
+        <div class="status-bar ai-blue-background"></div>
+        {% elsif x == pNum and lesson.num == page.num %}
+        {% assign smaller = 0 %}
+        <a class="ai-link" href="javascript:void(0)">#{{lesson.num}} {{lesson.title}}</a>
+        <br>
+        <div class="status-bar current-lesson"></div>
+        {% elsif x == pNum and smaller == 1 %}
+        <a class="ai-link" href="{{lesson.url}}">#{{lesson.num}} {{lesson.title}}</a>
+        <br>
+        <div class="status-bar ai-blue-background"></div>
+        {% else %}
+        <a class="ai-link" href="{{lesson.url}}">#{{lesson.num}} {{lesson.title}}</a>
+        <br>
+        <div class="status-bar lightgray-background"></div>
+        {% endif %}
+    {% endfor %}
+</div>
+
+<div id="mobile-sidebar">
+
+    <div id="sidebar-btn-container">
+        <div id="alternate-mobile-icon">Lessons</div>
+        <!-- <span class="hamburger-icon">
+            <span class="bar" id="top-bar"></span>
+            <span class="bar" id="middle-bar"></span>
+            <span class="bar" id="bottom-bar"></span>
+        </span> -->
+    </div>
+
+    <div id="hamburger-content">
+        {% for lesson in site.aiml %}
+        {% assign pNum = page.num | plus: 0 %}
+        {% assign x = lesson.num | plus: 0 %}
+        {% if x < pNum %} 
+            <div class="hamburger-item">
+                    <a class="ai-link" href="{{lesson.url}}">#{{lesson.num}} {{lesson.title}}</a>
+            </div>
+            {% else if x > pNum %}
+            <div class="hamburger-item">
+                <a class="ai-link" href="{{lesson.url}}">#{{lesson.num}} {{lesson.title}}</a>
+            </div>
+            {% else %}
+            <div class="hamburger-item">
+                <a class="ai-link" href="javascript:void(0)">#{{lesson.num}} {{lesson.title}}</a>
+            </div>
+            {% endif %}
+        {% endfor %}
+    </div>
+</div>
+
+<script>
+    function toggleSidebarDesktopToMobile() {
+        let desktop_sidebar = document.getElementById("desktop-sidebar");
+        let mobile_sidebar = document.getElementById("mobile-sidebar");
+        let window_width = window.innerWidth;
+        let left_margin = desktop_sidebar.offsetLeft;
+
+        left_margin = (window_width / 2.0 - 758) > 26 ? (window_width / 2.0 - 758) : 26;
+        if (window_width < 768) {
+            let hamburger_content = document.getElementById("hamburger-content");
+            if (hamburger_content.style.display != "flex") {
+                document.getElementById("hamburger-content").style.display = "none";
+            }
+            desktop_sidebar.style.display = "none";
+            mobile_sidebar.style.display = "inline-block";
+        } else {
+            desktop_sidebar.style.display = "inline-block";
+            mobile_sidebar.style.display = "none";
+            desktop_sidebar.style.marginLeft = left_margin + "px";
+        }

```

Comment:
This logic can be abstracted to a CSS class - that makes it a bit more readable than manually changing the style with JS, especially if it clashes with other CSS styles.

Reply:
I can't figure out how to control the left margin of the desktop sidebar in CSS. I want to do something like margin-left: max(calc(50vw -758px));
But this is a syntax error.

Reply:
What is the `max` doing here?

Reply:
I meant to do margin-left: max(calc(50vw-758), 10px);
Yea i think media queries would solve my issue